### PR TITLE
Install import

### PR DESCRIPTION
**Pull request by amueller**
You're importing GPyOpt in your ``setup.py``. If your user doesn't already have ``GPy`` installed, this will raise an error, because GPyOpt requires GPy.

You can either not import ``__version__`` in ``setup.py`` and hard-code the version here (and risk it becoming out-of-sync with ``__version__.py``), or use this fix (copied from sklearn). There is [several other solutions](https://stackoverflow.com/questions/17583443/what-is-the-correct-way-to-share-package-version-with-setup-py-and-the-package) to this somewhat common problem.

tldr; before this fix, ``pip install -e .`` raises an ``ImportError`` if ``GPy`` is not installed, with this fix it installs this dependency automatically.

### CODE REVIEW

**Comment by amueller** [comment]
``pip`` itself uses a regexp to search the ``__init__.py`` https://github.com/pypa/pip/blob/master/setup.py#L41 ... I think I'm going back to C++.

**Comment by javiergonzalezh** [comment]
Merged! Thanks so much for this.
